lib/validate.js

```diff
@@ -90,14 +93,21 @@ exports.usingBmap = (deviceFileDescriptor, options = {}) => {
  * });
  */
 exports.usingChecksum = (deviceFileDescriptor, options = {}) => {
+  const algorithms = options.imageChecksum
+    ? Object.keys(options.imageChecksum) : null;
+
   return checksum.calculateDeviceChecksum(deviceFileDescriptor, {
     imageSize: options.imageSize,
     progress: options.progress,
-    chunkSize: options.chunkSize
-  }).then((deviceChecksum) => {
-    if (options.imageChecksum !== deviceChecksum) {
-      throw errors.ValidationError;
-    }
+    chunkSize: options.chunkSize,
+    checksumAlgorithms: algorithms
+  }).then((results) => {
+
+    algorithms.forEach((algorithm) => {
+      if (options.imageChecksum[algorithm] !== results[algorithm]) {
```

Comment:
This means that if the user says he wants a certain algorithm (e.g: 'sha1'), but doesn't pass that checksum, we get a validation error?

I think we can either:

1. Throw an error if the passed checksums don't match the desired algorithm, or
2. Ignore a validation error if a checksum for that algorithm is missing

We should definitely have test cases for this as well.